jvm/src/Language/Java.hs

```diff
@@ -416,6 +427,9 @@ class (Interp (Uncurry a) ~ ty, SingI ty, IsReferenceType ty)
       => Reflect a ty where
   reflect :: a -> IO (J ty)
 
+  default reflect :: Coercible a ty => a -> IO (J ty)
+  reflect x = newLocalRef (jobject x)
+
```

Comment:
It is better to avoid using global references because it puts pressure on the Java heap, while the Haskell RTS is relaxed thinking there is no need to run a the GC.